pkg/engine/lifecycletest/pulumi_test.go

```diff
@@ -5550,204 +5550,168 @@ func TestContinueOnError(t *testing.T) {
 
-	// Run again, this time without erroring, to ensure the stack outputs are updated.
-	step = 3
-	snap, err = TestOp(Update).Run(p.GetProject(), p.GetTarget(t, snap), p.Options, false, p.BackendClient, nil)
-	assert.NoError(t, err)
-	validateSnapshot(snap, 1, resource.PropertyMap{
-		"first":  resource.NewProperty("step 3"),
-		"second": resource.NewProperty("step 3"),
-	})
+	// Run an update to create the resource
+	snap, err := TestOp(Update).Run(project, p.GetTarget(t, nil), p.Options, false, p.BackendClient, nil)
+	require.ErrorContains(t, err, "intentionally failed create")
+	require.NotNil(t, snap)
+	require.Equal(t, 5, len(snap.Resources)) // 3 resources + 2 providers
+	assert.Equal(t, resource.URN("urn:pulumi:test::test::pulumi:providers:pkgB::default"), snap.Resources[0].URN)
+	assert.Equal(t, resource.URN("urn:pulumi:test::test::pulumi:providers:pkgA::default"), snap.Resources[1].URN)
+	assert.Equal(t, resource.URN("urn:pulumi:test::test::pkgA:m:typA::independent1"), snap.Resources[2].URN)
+	assert.Equal(t, resource.URN("urn:pulumi:test::test::pkgA:m:typA::independent2"), snap.Resources[3].URN)
+	assert.Equal(t, resource.URN("urn:pulumi:test::test::pkgA:m:typA::independent3"), snap.Resources[4].URN)
 }
 
-// TestStackOutputsResourceError tests that previous stack outputs aren't deleted when an update fails
-// due to a resource operation error.
-func TestStackOutputsResourceError(t *testing.T) {
+func TestUpContinueOnErrorUpdate(t *testing.T) {
 	t.Parallel()
 
-	var step int
-
 	loaders := []*deploytest.ProviderLoader{
 		deploytest.NewProviderLoader("pkgA", semver.MustParse("1.0.0"), func() (plugin.Provider, error) {
+			return &deploytest.Provider{}, nil
+		}, deploytest.WithoutGrpc),
+		deploytest.NewProviderLoader("pkgB", semver.MustParse("1.0.0"), func() (plugin.Provider, error) {
 			return &deploytest.Provider{
-				CreateF: func(urn resource.URN, news resource.PropertyMap, timeout float64,
-					preview bool,
-				) (resource.ID, resource.PropertyMap, resource.Status, error) {
-					return "", nil, resource.StatusUnknown, errors.New("oh no")
+				UpdateF: func(urn resource.URN, id resource.ID,
+					oldInputs, oldOutputs, newInputs resource.PropertyMap,
+					timeout float64, ignoreChanges []string, preview bool,
+				) (resource.PropertyMap, resource.Status, error) {
+					return nil, resource.StatusOK, errors.New("intentionally failed update")
 				},
 			}, nil
-		}),
+		}, deploytest.WithoutGrpc),
 	}
 
+	ins := resource.NewPropertyMapFromMap(map[string]interface{}{
+		"foo": "bar",
+	})
+
+	// update as opposed to create for the failing resource
+	update := false
+
 	programF := deploytest.NewLanguageRuntimeF(func(_ plugin.RunInfo, monitor *deploytest.ResourceMonitor) error {
-		stackURN, _, _, _, err := monitor.RegisterResource(resource.RootStackType, "test", false)
+		_, _, _, _, err := monitor.RegisterResource("pkgB:m:typB", "failing", true, deploytest.ResourceOptions{
+			SupportsResultReporting: true,
```

Comment:
Same as above, we should have a version of this without SupportsResultReporting for testing old SDKs